src/Domain/Entity/ContactMediaMap.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Domain\Entity;
+
+use BBC\ProgrammesPagesService\Domain\Enumeration\ContactMediaEnum;
+use BBC\ProgrammesPagesService\Domain\ValueObject\ContactMedia;
+use InvalidArgumentException;
+
+class ContactMediaMap
+{
+    /**
+     * Example:
+     *  [
+     *      'email'    => [<ContactMedia>, <ContactMedia>, <ContactMedia>],
+     *      'facebook' => [<ContactMedia>],
+     *  ]
+     *
+     * @var ContactMedia[][]
+     */
+    private $contactsMapByMedia = [];
+
+    /**
+     * Example: ['reddit', 'email']
+     *
+     * @return string[]
+     */
+    public function getNamesOfUsedMedia(): array
+    {
+        return array_keys($this->contactsMapByMedia);
+    }
+
+    /**
+     * @return ContactMedia[]
+     */
+    public function getContactsByMedia(string $contactMediaType): array
+    {
+        $validsMedia = ContactMediaEnum::VALID_MEDIA;
+
+        if (!isset($validsMedia[$contactMediaType])) {
+            throw new InvalidArgumentException('Trying to get a contact detail with unknown type');
+        }
+
+        return $this->contactsMapByMedia[$contactMediaType] ?? [];
+    }
+
+    public function addContactMedia(ContactMedia $contactMedia): self
```

Comment:
Domain objects should be immutable. This means we don't have to worry about if they have been modified at any point in their lifecycle. I believe this object is the first that breaks that rule

Can you remove this `addContactMedia`, and instead construct the list of valid items within the Mapper.